### PR TITLE
Feature: New Evaluation Metrics

### PR DESCRIPTION
**Pull request by Gram21**
Adds Accuracy, [Phi Coefficient (aka MCC)](https://en.wikipedia.org/wiki/Phi_coefficient), and Specificity as evaluation metrics.
Slightly refactor the result-classes in tests-module.

- [x] New metrics for TLR
- [x] New metrics for Inconsistency Detection

### CODE REVIEW

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExtendedExplicitEvaluationResults.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        long a = truePositives + falsePositives;
+        long b = truePositives + falseNegatives;
+        long c = trueNegatives + falsePositives;
+        long d = trueNegatives + falseNegatives;
+        if (a == 0 || b == 0 || c == 0 || d == 0) {
+            return 0d;
+        }
+        long sumInDenominator = a * b * c * d;
+        double denominator = Math.sqrt(sumInDenominator);
```
Comment (dfuchss) [inline tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExtendedExplicitEvaluationResults.java:67]: Are there no better names ;)

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/integration/TraceabilityLinkRecoveryEvaluationIT.java [hunk 2]
```diff
@@ -202,12 +208,19 @@ private void checkResults(Project project, ArDoCoResult arDoCoResult) {
...
-                        .getRecall() + " is below the expected minimum value " + expectedResults.getRecall()), //
-                () -> Assertions.assertTrue(results.getF1() >= expectedResults.getF1(), "F1 " + results
-                        .getF1() + " is below the expected minimum value " + expectedResults.getF1()));
+                () -> Assertions.assertTrue(results.getPrecision() >= expectedResults.precision(), "Precision " + results
+                        .getPrecision() + " is below the expected minimum value " + expectedResults.precision()), //
+                () -> Assertions.assertTrue(results.getRecall() >= expectedResults.recall(), "Recall " + results
+                        .getRecall() + " is below the expected minimum value " + expectedResults.recall()), //
+                () -> Assertions.assertTrue(results.getF1() >= expectedResults.f1(), "F1 " + results
+                        .getF1() + " is below the expected minimum value " + expectedResults.f1()));
+        if (results instanceof ExtendedExplicitEvaluationResults<?> xxResults) {
```
Comment (dfuchss) [inline tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/integration/TraceabilityLinkRecoveryEvaluationIT.java:217]: Is there no better name for xxResults ;)

**Review by dfuchss — APPROVED** [review]
Minor comment, but nothing critical

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExtendedExplicitEvaluationResults.java (hunk 1 above)
Comment (Gram21) [reply tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExtendedExplicitEvaluationResults.java:67]: I can use w,x,y,z. They simply have no name themselves (only part of the formula). This is just to simplify the calculation.If you really want to use a better name, you can call them `sumOfTruePositivesAndFalsePositives`, but I really disliked the super lenghty if-clause. As they are only locally used there, I see no major problem

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/integration/TraceabilityLinkRecoveryEvaluationIT.java (hunk 2 above)
Comment (Gram21) [reply tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/integration/TraceabilityLinkRecoveryEvaluationIT.java:217]: Yes, `extendedExplicitEvaluationResults`, but then again the name was so long that the following code was too clustered. Maybe, we can rename it at some point

**Comment by Gram21** [comment]
I want to refactor and add additional code and extend this PR further. Therefore, expect future changes

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExpectedResults.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * @param recall         the expected recall
+ * @param f1             the expected F1 score
+ * @param accuracy       the expected accuracy
+ * @param phiCoefficient the expected Phi Coefficient
+ * @param specificity    the expected specificity
+ */
+public record ExpectedResults(double precision, double recall, double f1, double accuracy, double phiCoefficient, double specificity) {
+
+    public ExpectedResults(double precision, double recall, double f1) {
+        this(precision, recall, f1, .0, .0, 0.);
```
Comment (dfuchss) [inline tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/results/ExpectedResults.java:17]:
```suggestion
        this(precision, recall, f1, .0, .0, .0);
```

File: tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/EvaluationMetrics.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+        var a = tp.add(fp);
+        var b = tp.add(fn);
+        var c = tn.add(fp);
+        var d = tn.add(fn);
+        if (a.equals(BigDecimal.ZERO) || b.equals(BigDecimal.ZERO) || c.equals(BigDecimal.ZERO) || d.equals(BigDecimal.ZERO)) {
+            return 0d;
+        }
+
+        var sumInDenominator = a.multiply(b).multiply(c).multiply(d);
```
Comment (dfuchss) [inline tests/src/test/java/edu/kit/kastel/mcse/ardoco/core/tests/eval/EvaluationMetrics.java:147]: sumInDenominator  => product

**Review by dfuchss — APPROVED** [review]
Some minor comments but overall approve

**Pull request merged**